pkg/globalconfig/global_config.go

```diff
@@ -33,6 +32,8 @@ var (
 	DdevGlobalConfig GlobalConfig
 	// DdevProjectList is the list of all existing DDEV projects
 	DdevProjectList map[string]*ProjectInfo
+	// Old versions of DDEV stored the project list in the global config.
+	ddevLegacyGlobalConfig LegacyGlobalConfig
```

Comment:
Interesting technique, but couldn't we just allow the old ProjectList in current config, but not write it out?

Reply:
I assume that's what the `omitempty` does in https://github.com/ddev/ddev/pull/5926#pullrequestreview-1956645524?

If so, I have no problem doing that. I just wasn't aware that was an option.
When I tried doing it without `omitempty` it ended up with an empty map in the yaml file which is obviously misleading - for anyone who looks at that file it'd look like DDEV has incorrectly said there's no projects!